### PR TITLE
feat(workflow): Ignore subscription updates if org doesn't have access to incidents

### PR DESCRIPTION
**Pull request by iProgramStuff**
Currently, if a user downgrades their plan, they will continue to receive alerts as we continue to process the subscriptions as normal. This PR adds a feature check so that this doesn't happen.

### CODE REVIEW

File: src/sentry/incidents/subscription_processor.py [hunk 1]
```diff
@@ -106,6 +107,11 @@ def check_trigger_status(self, trigger, status):
         return incident_trigger is not None and incident_trigger.status == status.value
 
     def process_update(self, subscription_update):
+        if not features.has("organizations:incidents", self.subscription.project.organization):
```
Comment (wedamija) [inline src/sentry/incidents/subscription_processor.py:110]: @EvanPurkhiser Were you saying that access to transaction/performance metrics are gated behind an additional feature? If that's the case, we might need to check the dataset of the query and then check that flag as well.

Comment (EvanPurkhiser) [reply src/sentry/incidents/subscription_processor.py:110]: Weird I thought I left a comment but maybe I didn't submit.

Yes, I think we do need to gate on whether they have `incidents-performance` and if it's a transaction datatype query. We'll also have to potentially hide incidents that are of the datatype as well.

Comment (wedamija) [reply src/sentry/incidents/subscription_processor.py:110]: Ok, let's just add some simple logic here for now to check that extra feature flag if the dataset is transactions.

**Review by wedamija — APPROVED** [review]
lgtm, ci has an error though

**Pull request merged**